### PR TITLE
[macOS] Add support for SERP bridge with getInstallOriginVariant message

### PR DESCRIPTION
**Pull request by rachelmcr**
Task/Issue URL: https://app.asana.com/1/137249556945/project/1209825025475019/task/1215859134164545?focus=true
Tech Design URL: https://app.asana.com/1/137249556945/project/481882893211075/task/1216031354105425
CC:

### Description

Adds support for a new SERP bridge with a message so the Frontend can ask for an experiment's install-origin variant. Native only returns the variant; all search counting and storage stay on the SERP Frontend.

The new bridge is only used on macOS and the message is only supported in DMG builds.

### Testing Steps

1. Confirm implementation matches tech design.
2. Confirm unit test checks pass.

### Impact and Risks

None: Internal tooling

#### What could go wrong?

- Internal logic (user script, origin parsing, etc.) is unit tested according to spec.
- End-to-end messaging (request-response) will be tested from the frontend separately.

### Quality Considerations

- Privacy triage done from Frontend (covering all platforms); no pixels sent from native side.
- Security: 
   - SERP user script restricted to duckduckgo.com (or custom base URL set in debug menu).
   - Origin read-only from bundle xattr.

---
###### Internal references:
[Definition of Done](https://app.asana.com/0/1202500774821704/1207634633537039/f) | [Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552) | [Tech Design Template](https://app.asana.com/0/59792373528535/184709971311943)

---

> [!NOTE]
> **Low Risk**
> Feature is gated to Sparkle builds with origin-restricted messaging and read-only xattr; broad unit test coverage limits regression risk.
> 
> **Overview**
> Adds a **SERP content-scope bridge** so duckduckgo.com can call **`getInstallOriginVariant`** (plus **`handshake`**) and receive the experiment **`content`** segment from the install-origin xattr when eligibility passes. Native does not send pixels or persist SERP-side counts.
> 
> **AttributedMetric** gains **`InstallOriginParser`** and **`InstallOriginVariantResolver`** (≤28 ET days since install, entry `home`, campaign match, non-empty content). **`daysSinceInstalled`** now uses **`QuantisedTimePast.daysBetween`** for consistent ET day math.
> 
> **macOS** registers the new **`SERPInstallOriginUserScript`** on isolated content scope; **`installOriginEnabled`** and **`DefaultInstallOriginVariantProvider`** (xattr + install date) apply only on **Sparkle/DMG** builds—otherwise handshake reports install origin off and variant responses are **`nil`**. Message origins are restricted to the configured SERP host.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit c7dd6c70638aa60162b685952e8b8b5ccfda8476. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: macOS/DuckDuckGo/AttributedMetric/DefaultInstallOriginVariantProvider.swift [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        getXattr(named: AttributionXattr.origin, from: Bundle.main.bundlePath)
+    }),
+         installDateProvider: any AttributedMetricInstallDateProviding = AttributedMetricATBInstallDateProvider(),
+         dateProvider: any DateProviding = DefaultDateProvider()) {
+        self.originProvider = originProvider
+        self.installDateProvider = installDateProvider
+        self.dateProvider = dateProvider
+    }
+
+    func installOriginVariant(forCampaign campaign: String?) -> String? {
```
Comment (jotaemepereira) [inline macOS/DuckDuckGo/AttributedMetric/DefaultInstallOriginVariantProvider.swift:39]: 💡 This piece of code currently only calls the `InstallOriginVariantResolver`. I wonder if we could make an earlier return statement here with three cases: a) when the origin is nil, we return nil earlier, b) when the install date is nil, then we return nil earlier, c) when the requested campaign is either nil or empty, we return nil earlier.

So then the `InstallOriginVariantResolver.variant` call doesn’t need to check for any nil values, it already has all the right parameters it needs.

```swift
   func installOriginVariant(forCampaign campaign: String?) -> String? {
        guard let installDate = installDateProvider.installDate,
                let origin = originProvider.origin,
                let campaign,
                !campaign.isEmpty else {
            return nil
        }

        return InstallOriginVariantResolver.variant(
            origin: origin,
            requestedCampaign: campaign,
            installDate: installDate,
            referenceDate: dateProvider.now()
        )
    }
```

File: macOS/UnitTests/AttributedMetric/DefaultInstallOriginVariantProviderTests.swift [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import AttributedMetric
+import XCTest
+@testable import DuckDuckGo_Privacy_Browser
+import AttributedMetricTestsUtils
+
+final class DefaultInstallOriginVariantProviderTests: XCTestCase {
```
Comment (jotaemepereira) [inline macOS/UnitTests/AttributedMetric/DefaultInstallOriginVariantProviderTests.swift:24]: 🔍 With my suggestion above, I would include tests where each of the parameters is either nil (and also empty in the case of the requested campaign); we just return nil.

Then, for the `InstallOriginVariantResolver.variant` call, I would try to just mock it and make sure that if the other three are not nil, we just return what InstallOriginVariantResolver.variant returns. 

We shouldn’t test in this class what `InstallOriginVariantResolver.variant` does. We should have unit tests for that already.

File: SharedPackages/AttributedMetric/Sources/AttributedMetric/Models/InstallOriginVariantResolver.swift [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    /// 3. Origin campaign matches the requested campaign.
+    ///
+    /// Origin format: `funnel_entry_source_campaign_content` (see `InstallOriginParser`).
+    public static func variant(
+        origin: String?,
+        requestedCampaign: String?,
+        installDate: Date?,
+        referenceDate: Date
+    ) -> String? {
+        guard let origin,
```
Comment (jotaemepereira) [inline SharedPackages/AttributedMetric/Sources/AttributedMetric/Models/InstallOriginVariantResolver.swift]: 💡 With my suggested change this could be removed from this logic.

File: SharedPackages/SERPInstallOrigin/Sources/SERPInstallOrigin/SERPUserScript.swift [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import UserScript
+import WebKit
+
+public final class SERPUserScript: NSObject, Subfeature {
```
Comment (jotaemepereira) [inline SharedPackages/SERPInstallOrigin/Sources/SERPInstallOrigin/SERPUserScript.swift]: 💡 Should we rename this user script to be `SERPInstallOriginUserScript`? It will follow the same pattern as `SERPSettings,` and it is more concise; this name is broad. For example, I thought when I first saw it that this would include settings as well.

File: SharedPackages/SERPInstallOrigin/Sources/SERPInstallOrigin/SERPInstallOriginUserScript.swift [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+public final class SERPUserScript: NSObject, Subfeature {
+
+    public enum MessageName: String, CaseIterable {
+        case handshake
+        case getInstallOriginVariant
+    }
+
+    public enum DataModel {
+        public enum Platform: String, Codable {
+            case ios, macos
```
Comment (jotaemepereira) [inline SharedPackages/SERPInstallOrigin/Sources/SERPInstallOrigin/SERPInstallOriginUserScript.swift:32]: 🔍  Nitpicky: Thoughts on deriving the platform from compilation conditionals? It’s less prone to errors, instead of us manually setting the platform, we just use the `#if os(macOS)` and infer it from there.

File: SharedPackages/SERPInstallOrigin/Sources/SERPInstallOrigin/SERPUserScriptHandler.swift [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import UserScript
+
+protocol SERPUserScriptHandling {
```
Comment (jotaemepereira) [inline SharedPackages/SERPInstallOrigin/Sources/SERPInstallOrigin/SERPUserScriptHandler.swift]: Similar comment to the `SERPUserScript` we should rename this to follow the same pattern.

**Review by jotaemepereira — APPROVED** [review]
LGTM with some comments.

File: macOS/DuckDuckGo/AttributedMetric/DefaultInstallOriginVariantProvider.swift (hunk 1 above)
Comment (rachelmcr) [reply macOS/DuckDuckGo/AttributedMetric/DefaultInstallOriginVariantProvider.swift:39]: That's a nice improvement, thanks 👍

Comment (rachelmcr) [reply macOS/DuckDuckGo/AttributedMetric/DefaultInstallOriginVariantProvider.swift:39]: Updated this and related tests in eda6d0f.

File: SharedPackages/SERPInstallOrigin/Sources/SERPInstallOrigin/SERPUserScript.swift (hunk 4 above)
Comment (rachelmcr) [reply SharedPackages/SERPInstallOrigin/Sources/SERPInstallOrigin/SERPUserScript.swift]: Agreed, I missed this with renaming the package. Updated in b72d71f.

File: SharedPackages/SERPInstallOrigin/Sources/SERPInstallOrigin/SERPInstallOriginUserScript.swift (hunk 5 above)
Comment (rachelmcr) [reply SharedPackages/SERPInstallOrigin/Sources/SERPInstallOrigin/SERPInstallOriginUserScript.swift:32]: Good idea, updated in c7dd6c7.

**Comment by rachelmcr** [comment]
Confirmed with manual testing that the round-trip communication with the frontend works as expected, so this is ready to merge.
